### DISCUSSION TITLE
Support for Mercure (included with FrankenPHP) - a modern alternative to WebSockets

### DISCUSSION THREAD

**Discussion by benbjurstrom** [livewire/livewire#7529] [3. Feature ideas]
Hello,
Before I open a pull request I wanted to see if there was interest in adding support for [Mercure](https://mercure.rocks/) real-time events to Livewire. Mercure is a modern alternative to WebSockets that's a little simpler to setup in my opinion. It also ships included with [FrankenPHP](https://frankenphp.dev/) which has been getting some buzz recently.

I still need to write the tests and docs but you can find my proposed integration here: https://github.com/benbjurstrom/livewire/commit/13ab4a7bded7250dfda251eadb3954ab1e6c3c70
As you can see it's very lightweight.

I also put together a demo showing how to set things up in a real Livewire app. You can find the source code for the demo here: https://github.com/benbjurstrom/livewire-mercure-demo.

Looking forward to your feedback,
Thanks!

https://github.com/benbjurstrom/livewire-mercure-demo/assets/12499093/62f8372f-d79c-4e5a-ba46-aacb9792b428

**Comment by benbjurstrom** [comment]
Bumping this since Laravel just released Octane support for FrankenPHP workers.

https://blog.laravel.com/octane-frankenphp

**Reply by tpharaoh** [reply]
absolutely. this is the biggest thing IMO since Livewire itself. Amazing work @benbjurstrom 

FrankenPHP built in mercure
Livewire all the beautiful reactivity
Laravelnews said Laravel Broadcast will support mercure native

This is perfect.

Only thing missing is a Laravel Forge to deploy a franken server :)

**Reply by benbjurstrom** [reply]
> Only thing missing is a Laravel Forge to deploy a franken server :)

The Laravel team moves fast https://twitter.com/taylorotwell/status/1737504577430180275

<img width="600" alt="image" src="https://github.com/livewire/livewire/assets/12499093/c1781052-4ddb-4dbd-9969-82206fd15d6c">

**Comment by benbjurstrom** [comment]
As an update @calebporzio said he wants to keep this in userland until there's wider adoption and a clearer path to usage. Plus I suspect [Laravel Reverb](https://twitter.com/taylorotwell/status/1734925043669348807) will do some work in this space. The good news is Livewire makes it really easy to hook into events from an external script. 

I updated the demo repo linked above with an example implementation where the necessary javascript is loaded from [/resources/js/supportMercure.js](https://github.com/benbjurstrom/livewire-mercure-demo/blob/main/resources/js/supportMercure.js). No pull request or packages needed.

**Comment by simonbuehler** [comment]
hi, nice feature!
from an architectural pov when replacing websockets (Echo) with mercure:
imagine a pong game with two clients sending the move commands ( post over HTTP/2 , performance?) and the server maintains a state (ball & paddle positions ) and pushes state updates to the clients - what are your thoughts guys on performance vs websockets? Would you use mercure only for server push messaging?
